presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -456,19 +456,26 @@ void PrestoServer::initializeVeloxMemory() {
         asyncCacheSsdDisableFileCow);
   }
   const int64_t memoryBytes = memoryGb << 30;
-  std::shared_ptr<memory::MemoryAllocator> allocator;
   if (systemConfig->useMmapAllocator()) {
     memory::MmapAllocator::Options options;
     options.capacity = memoryBytes;
     options.useMmapArena = systemConfig->useMmapArena();
     options.mmapArenaCapacityRatio = systemConfig->mmapArenaCapacityRatio();
-    allocator = std::make_shared<memory::MmapAllocator>(options);
+    allocator_ = std::make_shared<memory::MmapAllocator>(options);
   } else {
-    allocator = memory::MemoryAllocator::createDefaultInstance();
+    allocator_ = memory::MemoryAllocator::createDefaultInstance();
   }
-  cache_ = std::make_shared<cache::AsyncDataCache>(
-      allocator, memoryBytes, std::move(ssd));
-  memory::MemoryAllocator::setDefaultInstance(cache_.get());
+  if (systemConfig->asyncDataCacheEnabled()) {
```

Comment:
Shall we skip initialize ssd cache if async data cache is not configured? We don't support to access SSD without async data cache for now. Thanks!